org/postgresql/geometric/PGline.java

```diff
@@ -17,38 +18,67 @@
 import java.sql.SQLException;
 
 /**
- * This implements a line consisting of two points.
+ * This implements a line represented by the linear equation Ax + By + C = 0
  *
- * Currently line is not yet implemented in the backend, but this class
- * ensures that when it's done were ready for it.
- */
+ **/
 public class PGline extends PGobject implements Serializable, Cloneable
 {
+
+    /**
+     * Coefficient of x
+     */
+    public double a;
+
+    /**
+     * Coefficient of y
+     */
+    public double b;
+
     /**
-     * These are the two points.
+     * Constant
      */
-    public PGpoint point[] = new PGpoint[2];
+    public double c;
 
     /**
-     * @param x1 coordinate for first point
-     * @param y1 coordinate for first point
-     * @param x2 coordinate for second point
-     * @param y2 coordinate for second point
+     * @param a coefficient of x
+     * @param b coefficient of y
+     * @param c constant
+     */
+    public PGline(double a, double b, double c) {
+        this();
+        this.a = a;
+        this.b = b;
+        this.c = c;
+    }
+
+    /**
+     * @param x1 coordinate for first point on the line
+     * @param y1 coordinate for first point on the line
+     * @param x2 coordinate for second point on the line
+     * @param y2 coordinate for second point on the line
      */
     public PGline(double x1, double y1, double x2, double y2)
     {
-        this(new PGpoint(x1, y1), new PGpoint(x2, y2));
+        this();
+        double a = (y2 - y1) / (x2 - x1);
```

Comment:
The div-by-zero case is handled by checking to see if the points have the same X coordinate and just using the coefficients for a vertical line (A=-1 and B=0) if they are equivalent.

The case where we have NaN or infinity values passed in... I'm not sure if there is a proper way to handle those cases.  We could test the values and throw an our own exception... or let an exception be thrown from the operation... or maybe there's another option that I'm not thinking of at the moment.

I guess, alternatively, we could punt on this particular constructor completely and break compatibility.  I don't know if it was ever working to begin with :smile:  But since postgresql allows you to specify two points on the line as a value, I figured maintaining a constructor to do this would be good.